lib/bootbasetest.pm

```diff
@@ -10,19 +10,17 @@ sub post_fail_hook {
         record_info 'Seems that the display manager failed';
     }
 
-    # if we found a shell, we do not need the memory dump
-    if (!(match_has_tag('emergency-shell') or match_has_tag('emergency-mode'))) {
-        die "save_memory_dump not implemented, no way to save memory_dump" unless check_var('BACKEND', 'qemu');
-        select_console 'root-console';
-        diag 'Save memory dump to debug bootup problems, e.g. for bsc#1005313';
-        save_memory_dump;
-    }
-
     # crosscheck for text login on tty1
     select_console 'root-console';
 
     # collect and upload some stuff
-    export_logs();
+    opensusebasetest->export_logs();
+
+    # Dump memory on qemu backend
+    if (check_var('BACKEND', 'qemu')) {
+        diag 'Save memory dump to debug bootup problems, e.g. for bsc#1005313';
+        save_memory_dump;
```

Comment:
saving the memory dump can be very heavy on the stored assets so I would do that *only* if necessary. Your commit says "Rework" but no better motivation. Better we try to do this only as a "last resort" whenever we found neither a login prompt nor a shell.

Reply:
Should we guard it with a `DEBUG_DUMP_MEMORY` var?

The rework is because current way is broken and does not make much sense. And save_memory_dump is broken anyway, at least on aarch64. You may prefer the word `simplify` or `sanitize`? Hard to explain more as it is just a rework.

Reply:
I would really prefer to catch memory dumps automatically when there is no better way for debugging available. What is the problem with evaluating all alternatives as I suggested?

Reply:
Is it possible to have `select_console` not fatal? Otherwise, it will be very difficult.